tests/controllers/test_rss_controller.py

```diff
@@ -141,10 +182,56 @@ def test_try_parse_feed_with_beautiful_soup(feed_input: str, is_url: bool, enabl
 
 @pytest.mark.parametrize("enable_cache", [True, False])
 def test_enable_disable_cache(enable_cache):
-    feed_input = "https://rss.nytimes.com/services/xml/rss/nyt/HomePage.xml"
-    rss_controller = RSSController(feed_input=feed_input, enable_cache=enable_cache)
+    controller = RSSController(feed_input=test_urls, enable_cache=enable_cache)
 
     if enable_cache:
-        assert rss_controller._session
+        assert controller._session
     else:
-        assert not rss_controller._session
+        assert not controller._session
+
+
+def test_parse_feeds(mock_feed: feedparser.FeedParserDict):
+    feed_input = test_urls[0]
+    controller = RSSController(feed_input=feed_input, enable_cache=False)
+
+    with patch("morpheus.controllers.rss_controller.feedparser.parse") as mock_feedparser_parse:
+
+        mock_feedparser_parse.return_value = mock_feed
+
+        with patch.object(controller, '_try_parse_feed') as mock_try_parse_feed:
+            dataframes_generator = controller.parse_feeds()
+            next(dataframes_generator, None)
+            feed_stats: FeedStats = controller._feed_stats_dict[feed_input]
+            assert feed_stats.last_try_result == "Success"
+            assert feed_stats.failure_count == 0
+            assert feed_stats.success_count == 1
+
+            # Raise exception to test failure scenario
+            mock_try_parse_feed.side_effect = Exception("SampleException")
+            dataframes_generator = controller.parse_feeds()
+            next(dataframes_generator, None)
+
+            feed_stats: FeedStats = controller._feed_stats_dict[feed_input]
+            assert feed_stats.last_try_result == "Failure"
+            assert feed_stats.failure_count == 1
+            assert feed_stats.success_count == 1
+
+            # Skip trying until cooldown period is met.
+            dataframes_generator = controller.parse_feeds()
+            next(dataframes_generator, None)
+
+            feed_stats: FeedStats = controller._feed_stats_dict[feed_input]
+            assert feed_stats.last_try_result == "Failure"
+            assert feed_stats.failure_count == 1
+            assert feed_stats.success_count == 1
+
+            # Resume trying after cooldown period
+            with patch("time.time", return_value=time.time() + controller._cooldown_interval):
```

Comment:
Same issue here, although is a bit different since we're not testing _cooldown_interval at all. I'd just make this a static value for the test and indicate that it depends on the cooldown interval specified for the stage.